rules/xcodeproj.bzl

```diff
@@ -43,6 +43,18 @@ def _srcs_info_build_files(ctx):
 
     return [path]
 
+def _xcodeproj_aspect_collect_hmap_paths(deps, target, ctx):
+    hmap_paths = []
```

Comment:
You could instead look for the HeaderMapInfo in the providers for the dep. Also, for correctness, you should probably keep the private hmap flags in there too. That doesn't break anything, does it? 

```
def _xcodeproj_aspect_collect_hmap_paths(deps, target, ctx):
    hmap_paths = []
    kind = ctx.rule.kind
    if kind == "objc_library" or kind == "swift_library":
        for dep in deps:
            if dep and HeaderMapInfo in dep:  # type(dep) == "Target":
                path = ctx.expand_location("$(execpath :%s)" % dep.label.name)
                hmap_paths.append(path)
    return hmap_paths
```

Reply:
You may also want to leave a comment explaining that we are doing this, because the library.bzl macro generates hmaps and refers to them in the value of the `-I` option.

Reply:
So regarding the `private_hmap ` one it is because `library.bzl` only has it under `-iquote` instead of `-I`, if we include this then private headers will be exposed as public ones? 
Now think again what I realize is that if someone did this, then build will not pass (assuming what bazel did is source of truth), then it does not matter whether I match the flags here or not. Therefore in the latest commit I will not do this check to make code cleaner

Reply:
`if we include this then private headers will be exposed as public ones?` Bazel's clang invocation does reference the private hmaps. We aren't exposing private headers per se, we are just making sure that the way we build these in bazel is the same as the way we do it in xcode.

It seems like library.bzl adds private_hmap as both `-I` and `-iquote`.